cmd/zed/manage/lakemanage/branch.go

```diff
@@ -0,0 +1,104 @@
+package lakemanage
+
+import (
+	"context"
+	"time"
+
+	"github.com/brimdata/zed/api"
+	lakeapi "github.com/brimdata/zed/lake/api"
+	"github.com/brimdata/zed/lake/pools"
+	"github.com/brimdata/zed/lakeparse"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/zap"
+)
+
+type branch struct {
+	config Config
+	lake   lakeapi.Interface
+	logger *zap.Logger
+	pool   *pools.Config
+	name   string
+	tasks  []func(context.Context) (*time.Time, error)
+}
+
+func newBranch(branchName string, pool *pools.Config, lake lakeapi.Interface, config Config, logger *zap.Logger) *branch {
+	b := &branch{
+		config: config,
+		lake:   lake,
+		logger: logger.Named("pool").With(
+			zap.String("name", pool.Name),
+			zap.Stringer("id", pool.ID),
+			zap.String("branch", branchName),
+		),
+		pool: pool,
+		name: branchName,
+	}
+	if !config.Compact.Disabled {
+		b.tasks = append(b.tasks, b.compact)
+	}
+	if config.Index.Enabled() {
+		b.tasks = append(b.tasks, b.index)
+	}
+	return b
+}
+
+func (b *branch) compact(ctx context.Context) (*time.Time, error) {
+	b.logger.Debug("compaction started")
+	head := lakeparse.Commitish{Pool: b.pool.Name, Branch: b.name}
+	it, err := NewPoolDataIterator(ctx, b.lake, &head, b.pool.Layout)
+	if err != nil {
+		return nil, err
+	}
+	var nextcold *time.Time
+	ch := make(chan Run)
+	go func() {
+		nextcold, err = CompactionScan(ctx, it, b.pool, b.config.Compact.ColdThreshold, ch)
+		close(ch)
+	}()
+	var found int
+	var compacted int
+	for run := range ch {
+		commit, err := b.lake.Compact(ctx, b.pool.ID, head.Branch, run.ObjectIDs(), api.CommitMessage{})
+		if err != nil {
+			return nil, err
+		}
+		found++
+		compacted += len(run.Objects)
+		b.logger.Debug("compacted", zap.Stringer("commit", commit), zap.Int("objects_compacted", len(run.Objects)))
+	}
+	if compacted == 0 {
+		b.logger.Debug("compaction completed", zap.Int("runs_found", found), zap.Int("objects_compacted", compacted))
+	} else {
+		b.logger.Info("compaction completed", zap.Int("runs_found", found), zap.Int("objects_compacted", compacted))
+	}
```

Comment:
But to be honest, I think we should log everything at the info level.
```suggestion
	level := zap.InfoLevel
	if compacted == 0 {
		level = zap.DebugLevel
	}
	b.logger.Log(level, "compaction completed", zap.Int("runs_found", found), zap.Int("objects_compacted", compacted))
```

Reply:
I can dig it 👍🏻